### ISSUE TITLE
Namespace conflict with campaign core component

### ISSUE THREAD

**Issue by jofranz** [systopia/de.systopia.campaign#110]
Since CiviCRM version 5.63.0

References:
- GitLab-Issue [Proposal: Create stub extensions for civicrm core + all components](https://lab.civicrm.org/dev/core/-/issues/3961)
   - included: [namespace communication](https://lab.civicrm.org/dev/core/-/issues/3961#note_92824)
- GitLab-Issue: [Meta: Create list of items for moving a core component to an extension](https://lab.civicrm.org/dev/core/-/issues/3089)
- GitHub-PR: [dev/core#3961 Create stub extensions for all components](https://github.com/civicrm/civicrm-core/pull/26036)
- GitHub-PR: [dev/core#3961 - Modify psr-0 namespaces of component extensions to match core](https://github.com/civicrm/civicrm-core/pull/26256)

_ref: 21638_

**Comment by bjendres** [comment]
I guess the plan would be

1. communicate that this is an issue (e.g , users upgrade to CiviCRM ``5.63.0``
1. Find some funding if this turns out to be a lot of work
1. change namespace, e.g. to ``campaignmanager``. That might require renaming all/most classes, but maybe there's a way around it
1. then see what an extension upgrade looks like and see how we can make it as little painful as possible
1. Find some funding if this turns out to be a lot of work

**Comment by jensschuppe** [comment]
I'd suggest merging the last pending PR and releasing a `1.3.0`, stating in the release notes that there's an incompatibility with Core `5.63.0`+ ~~and that the next version will have to be renamed~~.

~~Maybe this will involve renaming the repository as well, and of course the extension directory entry: https://civicrm.org/extensions/campaign-manager.~~

**Comment by sluc23** [comment]
we can help on renaming and testing the extensions.. we just discovered this issue when we tried to migrate to Civi5.63.x whith *Campaing Manager* ext installed, and we are locked now until this is fixed..

**Comment by bjendres** [comment]
> we can help on renaming and testing the extensions.. 

That would be great! The current draft is in #111, let us know if you can find any issues with it. @jensschuppe, who has looked into the issue, is now on holiday, but I might be able to help you with minor things...

**Comment by sluc23** [comment]
hmm PR#111 looks still very [WIP].. some miss-matches in the new class names are not working.. i.e [this](https://github.com/systopia/de.systopia.campaign/pull/111/files#diff-0a61e2d6fd5a8170e0f5c86b72b9e7586ede90a096a02de31603e57bbcb4c897R16) where class `CRM_CampaignTree_CampaignManager_Page_Dashboard` is in folder `CRM/CampaignManager/CampaignTree/Page/Dashboard.php`, so it's not found..

@bjendres we need to move forwards with this fix, we are blocked on CiviCRM upgrades for all our sites due to this.. what's the best approach? would you like us re-work on @jensschuppe PR or create our own PR while he's on vacations?

**Comment by bjendres** [comment]
@sluc23 and I talked briefly, and he would have a go at fixing the issue in the last comment, and we'll see where we are with the compatibility on Monday (Aug 14th) together with @jensschuppe.

**Comment by jofranz** [comment]
@sluc23 Thanks for giving us this hint! That indeed was quite helpful and saved us debugging time.

We increased our internal priority as this is blocking potential security updates. I pushed my this change to #111. Would you mind give it a try?

**Comment by sluc23** [comment]
@jofranz I think this PR needs more work.. 
- tpls are in the wrong folders (it should be `/templates/CRM/CampaignManager/CampaignTree/Form/Search.tpl` for example for class renamed `CRM_CampaignManager_CampaignTree_Form_Search`)
- namespace need to be redefined as well in hooks [here](https://github.com/systopia/de.systopia.campaign/blob/renameNamespace/campaign.php#L80)
- AngularJS pages are not loading in my dev.., but i'm double checking, it could be a local issue.

I can propose a different approach that is to start a brand new extension called `campaignmanager` with the latest `civix` and CiviCRM coding standards and move the features there.. 
Coding standards in this extension is quite outdated and could me much improved
It could take some time, but maybe now it's the time to move to that direction to avoid future conflicts in further Civi releases.

We can take the lead on this development and allocate time to make it happen.

What do you think?

**Comment by jensschuppe** [comment]
As discussed with @bjendres, we'd like to keep this extension's name and repository with as little adjustments as possible for a current version working with latest CiviCRM Core versions. @sluc23 offered to rework the extension's old code base to meet modern standards in a fork and maybe merge into here as a version 2.0, see [milestone](https://github.com/systopia/de.systopia.campaign/milestone/7).

I made some more adjustments in the PR #111 for getting the current state to work with current CiviCRM and PHP versions and had a quick look at the UI, seeing no fatal errors and the AngularJS UI working as before. @sluc23 would you be able to confirm this PR fixes the issue?
